Rakefile

```diff
@@ -1,29 +1,122 @@
-require 'rspec/core/rake_task'
 require 'fileutils'
 require 'rototiller'
+require "yard"
+require "rubocop/rake_task"
+require "flog_task"
+require "flay_task"
+require "roodi_task"
+require "rubycritic/rake_task"
+
+YARD_DIR = "doc".freeze
+DOCS_DIR = "docs".freeze
 
 task :default do
-  sh %{rake -T}
+  sh %(rake -T)
+end
+
+namespace :gem do
+  require "bundler/gem_tasks"
 end
 
-# temporary backwards compat
-task :test => :'test:unit'
-task :acceptance => :'test:acceptance'
+task yard: :"docs:yard"
+
+# rubocop:disable Metrics/BlockLength
+namespace :docs do
+  # docs:yard task
+  YARD::Rake::YardocTask.new
+
+  desc "Clean/remove the generated YARD Documentation cache"
+  task :clean do
+    original_dir = Dir.pwd
+    Dir.chdir(File.expand_path(File.dirname(__FILE__)))
```

Comment:
Why are we changing directories? We can remove a directory without cd'ing into it.

Reply:
randell asked the same thing.  i need to comment this, because i keep forgetting myself.  i think the idea is that rake can be run from any dir below project root, so we never know what dir we're in.  but we know _where_ the rakefile is with `__FILE__`.  but i guess we don't have to cd to the dir.  i can just absolute path the yarddir with the __FILE__ path.